tests/unit/test_core/test_link.py

```diff
@@ -209,14 +209,15 @@ def test_get_next_available_tag(self, controller):
         link = Link(self.iface1, self.iface2)
         tag = link.get_next_available_tag(controller, "link_id")
         next_tag = link.get_next_available_tag(controller, "link_id")
-
+        assert tag == 1
         assert tag != next_tag
 
-    def test_get_tag_multiple_calls(self, controller):
-        """Test get next available tags returns different tags"""
+    def test_get_next_available_tag_reverse(self, controller):
+        """Test get last available tags returns different tags"""
         link = Link(self.iface1, self.iface2)
-        tag = link.get_next_available_tag(controller, "link_id")
-        next_tag = link.get_next_available_tag(controller, "link_id")
+        tag = link.get_next_available_tag(controller, "link_id", True)
+        next_tag = link.get_next_available_tag(controller, "link_id", True)
+        assert tag == 4095
```

Comment:
that's great, let's also assert `next_tag == 4094`, since that also cover part of the expected behavior when `take_last=True`